qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -0,0 +1,828 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Analysis class for curve fitting.
+"""
+# pylint: disable=invalid-name
+
+import functools
+from collections import defaultdict
+from typing import Any, NamedTuple, Dict, List, Optional, Tuple, Callable, Union
+
+import numpy as np
+import scipy.optimize as opt
+from qiskit.exceptions import QiskitError
+
+from qiskit_experiments.base_analysis import BaseAnalysis
+from qiskit_experiments.data_processing import DataProcessor
+from qiskit_experiments.data_processing.exceptions import DataProcessorError
+from qiskit_experiments.experiment_data import AnalysisResult, ExperimentData
+
+# Description of data properties for single curve entry
+SeriesDef = NamedTuple(
+    "SeriesDef",
+    [
+        ("name", str),
+        ("param_names", List[str]),
+        ("fit_func_index", int),
+        ("filter_kwargs", Optional[Dict[str, Any]]),
+        ("data_option_keys", Optional[List[str]]),
+    ],
+)
+
+# Human readable data set for single curve entry
+CurveEntry = NamedTuple(
+    "CurveEntry",
+    [
+        ("curve_name", str),
+        ("x_values", np.ndarray),
+        ("y_values", np.ndarray),
+        ("y_sigmas", np.ndarray),
+        ("metadata", dict),
+    ],
+)
+
+
+def scipy_curve_fit_wrapper(
```

Comment:
This is from the old discussion with Zlatko. Actually this `scipy_curve_fit_wrapper` is the renamed version of `curve_fit`, and attached to the class attribute `__base_fitter__`. Because he proposed to use another fit module rather than scipy (I think we will get similar request from community, at some point), renaming this function with `scipy` prefix seems reasonable to me.

Reply:
Maybe we can directly import fit function from the original place, to keep the git blame.